backend/src/services/organizationService.ts

```diff
@@ -284,20 +300,31 @@ export default class OrganizationService extends LoggerBase {
         repoOptions,
       )
 
-      // check website before creating the secondary org
-      if (
-        payload.secondary.website &&
-        payload.secondary.website === organization.website &&
-        !payload.primary.website
-      ) {
-        // set primary website to null before creating the secondary org
-        await OrganizationRepository.update(
-          organizationId,
-          { website: null },
-          repoOptions,
-          false,
-          false,
-        )
+      // check domains before creating the secondary org
+      const orgDomains = organization.identities
+        .filter((i) => i.verified && i.type === OrganizationIdentityType.PRIMARY_DOMAIN)
+        .map((i) => i.value)
+
+      const primaryDomains = payload.primary.identities
+        .filter((i) => i.type === OrganizationIdentityType.PRIMARY_DOMAIN)
+        .map((i) => i.value)
+      const secondaryDomains = payload.secondary.identities
+        .filter((i) => i.type === OrganizationIdentityType.PRIMARY_DOMAIN)
+        .map((i) => i.value)
+
+      // check if any domain is set to the org and came from secondary and is not in primary
+      // because we need to remove it from org and create it to the unmerged org
+      for (const domain of secondaryDomains) {
+        if (orgDomains.include(domain) && !primaryDomains.includes(domain)) {
+          const identity = organization.identities.find(
+            (i) => i.type === OrganizationIdentityType.PRIMARY_DOMAIN && i.value === domain,
+          )
+          await OrganizationRepository.removeIdentitiesFromOrganization(
+            organizationId,
+            [identity],
+            repoOptions,
+          )
+        }
       }
```

Comment:
The problem is that in the db we probably don't have the verified flag yet and there is no easy way to add it...
So the unmerge backup won't have the correct structure - that's why in the repo layer I kinda try to recreate the correct structure (with value, type instead of name) but the verified flag is not set...

Reply:
That's why I only check verified flag for identities coming from org that came from db.